apis/v1alpha1/httproute_types.go

```diff
@@ -377,70 +396,116 @@ type HTTPRequestHeaderFilter struct {
 	//   GET /foo HTTP/1.1
 	//   My-Header2: DEF
 	//
-	// Support: extended?
+	// Support: Extended
+	// +kubebuilder:validation:MaxItems=16
 	Remove []string `json:"remove"`
-
-	// TODO
 }
 
-// HTTPRequestMirrorFilter defines configuration for the
-// RequestMirror filter.
+// HTTPRequestMirrorFilter defines configuration for the RequestMirror filter.
 type HTTPRequestMirrorFilter struct {
-	// TargetRef is an object reference to forward matched requests to.
-	// The resource may be "services" (omit or use the empty string for the
-	// group), or an implementation may support other resources (for
-	// example, resource "myroutetargets" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "services".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
+	// ServiceName refers to the name of the Service to mirror matched requests
+	// to. When specified, this takes the place of BackendRef. If both
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this
+	// Route to true.
 	//
-	// Support: Core (Kubernetes Services)
-	// Support: Implementation-specific (Other resource types)
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
```

Comment:
// +kubebuilder:validation:Minimum=1

Reply:
Why would we add a minLength validation of 1 on an optional field?

Reply:
My bad, I missed that this is an optional field.